Don't overwrite `self.teams` upon parsing responders in OpsGenie integration

lstyles opened the pull request:
## Description

<!--
Provide a description for your pull request. Note any breaking changes.
-->

## Checklist

<!--
The following checklist items must be completed before PRs can be merged. 
-->

- [x] I have reviewed the [contributing guidelines](https://github.com/jertel/elastalert2/blob/master/CONTRIBUTING.md).
- [x] I have included unit tests for my changes or additions.
- [x] I have successfully run `make test-docker` with my changes.
- [x] I have manually tested all relevant modes of the change in this PR.
- [x] I have updated the [documentation](https://elastalert2.readthedocs.io).
- [x] I have updated the [changelog](https://github.com/jertel/elastalert2/blob/master/CHANGELOG.md).


## Questions or Comments

<!--
If any of the checklist items do not apply, note the reasoning for each. If you're simply
upgrading a library version, you do not need to explain why the docs or unit tests checklist
items are not checked, however the changelog should be updated to reflect the new version.

If you have questions about completing this PR, or about the process, note them here.

If you are not ready for this PR to be reviewed please mention that here.
-->

jertel commented:
Thanks for submitting this. There are already extensive unit tests for this alerter, so adjusting one of them to send two alerts and then assert that the second one was unaffected by the first should (hopefully) be low-effort.

lstyles commented:
Thanks. Updated, but the changes needed were a bit more extensive than I would have liked.

I've added a test for this particular issue, but had to make some other changes to tidy things up a bit. Some of the tests were checking for (what I think) were incorrect values and some others were checking properties on the alert object instead of mocked post request.

Let me know what you think. Apologies if it is a mess, I've not done any Python dev prior to this.

lstyles commented on elastalert/alerters/opsgenie.py:
```diff
@@ -35,25 +35,29 @@ def __init__(self, *args):
-                else:
-                    formated_responders = default_responders
-            responders = formated_responders
+        if responders is None:
+            return None
+        if responder_args is None:
+            responder_args = dict()
```
The top two lines are potentially redundant.

The bottom two lines fix a scenario where `responders` contains a placeholder, but `responders_args` isn't configured in a rule at all. Previously, a text with placeholder would end up as a rule. Now, a warning will be shown and `default_responders` will be used instead.

lstyles commented:
The two new tests could probably be named to better describe what they're doing:
- preserve_rule_config_upon_parsing_teams
- preserve_rule_config_upon_parsing_recipients
or something like that

nsano-rururu commented:
The next version, 2.21.0, will allow both the old setting name and the correct wording. However, if you use the old setting name, how about informing the user that the old setting value will be deleted when the logger reaches 2.25.0 every time an alert notification is sent?

nsano-rururu commented:
For warning loggers, use elastalert_logger.warning.

example
https://github.com/jertel/elastalert2/pull/793/files

lstyles commented:
I've put something together but on alerter initialisation instead of alert send. Deprecation logic feels more in place there vs having it in the alert method.

nsano-rururu commented:
Update CHANGELOG.md.

nsano-rururu commented:
@jertel 

Any other comments?

jertel commented:
Looks good, thank you both. The schema.yaml is missing those two properties, as well as other OpsGenie properties. But that can wait for another PR since it's not directly related to this.

lstyles commented:
https://github.com/jertel/elastalert2/pull/1543 schema update as a separate PR (1 - to not block this one, 2 - because I'm not 100% sure it's correct)
